From a7a24d491e2b339abf7240096a9a906b7a3a8145 Mon Sep 17 00:00:00 2001
From: Tally Jacobs <tally.jacobs@tandemg.com>
Date: Wed, 20 Oct 2021 10:10:14 +0300
Subject: [PATCH] [SW-55007] comms: extend binning conf support

Controlling EDMA binning conf from LKD
to allow flexebility during power on tests.
The message is set to preboot and passed to u-boot
since the data is required at early u-boot boot stages.

Change-Id: I55febf3ddc36e56235ea698cfce1340ebc9bd164
Signed-off-by: Tally Jacobs <tally.jacobs@tandemg.com>
---

diff --git a/include/preboot.h b/include/preboot.h
index e0a0970..5da5aa2 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -84,6 +84,7 @@
 	uint64_t tpc;
 	uint32_t dec;
 	uint32_t hbm;
+	uint32_t edma;
 };
 
 struct hl_binning_conf {
diff --git a/lib/preboot/module_lkd.c b/lib/preboot/module_lkd.c
index 7ddfc8e..181077f 100644
--- a/lib/preboot/module_lkd.c
+++ b/lib/preboot/module_lkd.c
@@ -136,6 +136,7 @@
 	masks.tpc = mcomms->tpc_binning_conf;
 	masks.dec = mcomms->dec_binning_conf;
 	masks.hbm = mcomms->hbm_binning_conf;
+	masks.edma = mcomms->edma_binning_conf;
 	preboot_set_binning_conf(&masks);
 
 	return 0;
